src/main/java/net/minecraftforge/server/permission/nodes/PermissionDynamic.java

```diff
@@ -17,30 +17,20 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
-package net.minecraftforge.server.permission.context;
+package net.minecraftforge.server.permission.nodes;
 
-import com.google.common.base.Preconditions;
-import net.minecraft.world.entity.player.Player;
-import net.minecraft.world.level.Level;
+import net.minecraft.util.StringRepresentable;
 
-public class PlayerContext extends Context
+/**
+ * <p>PermissionDynamics can be seen like Block Properties.
+ * An example usage might be a dimension inside a check for building permissions:
+ * "modid.build.allowed[dimension=overworld]" </p>
+ * <p>Note: The string representation is just an example and may be different for other PermissionHandlers</p>
+ * @param <T> type of the PermissionDynamic, must implement {@link StringRepresentable}
+ */
+public record PermissionDynamic<T extends StringRepresentable> (Class<T> typeToken, String name)
```

Comment:
Don't get me wrong, but I don't find the documentation clear.
A better idea would be
```
Represents a key that can be used to build a {@link PermissionDynamicContext}.

<p>Keys, along with their associated values, can be used to provide additional context for a permission handler in determining whether to grant permission for an actor and a specific node.</p>

<p>As an example usage, a dimension context key could be used inside a building permission check to ensure that the actor can build given those constraints.</p>
```

I'd also rename this class to `PermissionDynamicContextKey` to make it easier to understand that it doesn't store a value. Also, it could be nice to indicate that there is no restriction on the values of the value (pun not intended), since that's the exact opposite of how block properties behave.